src/testrun.rs

```diff
@@ -46,63 +48,145 @@ impl Display for Outcome {
     }
 }
 
-#[derive(Clone, Debug)]
+static FRAMEWORKS: LazyLock<Vec<(&str, Framework)>> = LazyLock::new(|| {
+    vec![
+        ("pytest", Framework::Pytest),
+        ("vitest", Framework::Vitest),
+        ("jest", Framework::Jest),
+        ("phpunit", Framework::PHPUnit),
+    ]
+});
+static EXTENSIONS: LazyLock<Vec<(&str, Framework)>> =
+    LazyLock::new(|| vec![(".py", Framework::Pytest), (".php", Framework::PHPUnit)]);
+
+fn check_substring_before_word_boundary(string: &str, substring: &str) -> bool {
+    if let Some((_, suffix)) = string.to_lowercase().split_once(substring) {
+        match suffix.chars().next() {
+            None => {
+                return true;
+            }
+            Some(first_char) => {
+                if !first_char.is_alphanumeric() {
+                    return true;
+                }
+            }
+        }
```

Comment:
```suggestion
        return suffix.chars().next().map_or(true, |first_char| !first_char.is_alphanumeric());
```

I think that should do the trick as well. `Option` has a ton of useful functional combinators.